src/middlewared/middlewared/plugins/smb.py

```diff
@@ -164,6 +289,11 @@ def check_codec(encoding):
             except (ValueError, TypeError):
                 verrors.add(f'smb_update.{i}', 'Not a valid mask')
 
+        if new['admin_group']:
```

Comment:
Should we make sure old admin_group is different so we dont call `add_admin_group` every smb update?